core/src/main/java/org/bitcoinj/core/Message.java

```diff
@@ -84,6 +84,10 @@ protected Message(NetworkParameters params, ByteBuffer payload, MessageSerialize
         this.payload = null;
     }
 
+    protected Message(ByteBuffer payload) throws ProtocolException {
+        this(null, payload, DummySerializer.DEFAULT);
+    }
+
```

Comment:
It basically means there is no serializer. We should actually pass `null`, but I didn't want to mix concepts.

Yes, messages should not know about their serializer. Currently they do, because they construct their child messages using the `MessageSerializer` interface. Makes no sense, they can as well construct it directly – see #2928.

Also some messages use just the protocol version to distinguish between an old and a new format (the message changed with some BIP). I've not yet a good solution for that, other than removing the serialization code entirely.